### DISCUSSION TITLE
How this tool decides target being RGB vs RGBA?

### DISCUSSION THREAD

**Discussion by VanderSP** [zeux/meshoptimizer#663] [General]
I want to know because I tend to use png as possible in my meshes.. but many times im not using alpha information... if this tool targets RGBA just because of png being the input, would be a waste.

**Comment by zeux** [comment]
What "target" are you asking about? glTF materials contain blend information; gltfpack may remove it if textures associated with it have no alpha channel as an optimization step.

**Reply by VanderSP** [reply]
Ah thanks for fast reply, well... i convert my meshes using your tool... but while doing some separated texture conversions for other parts using toktx tool, i observed that if my input is jpg for example, and i target rgba instead rgb, it wastes 200kb more of useless channel, so i wanted to know if your tool do this in a smartway even if gltf textures are png on mesh (even without alpha, just for being lossless as possible), since we can´t choose the target for each texture

I would like to also ask you a basisu encoder question, as i encode in device some stuff, it feels like a simplified toktx... so i can´t access those super fancy extra conversions like --zcmp / --uastc_rdo_l or they are hardcoded in a smart way in they encoder?

Thanks by your hardwork Zeux!

**Reply by zeux** [reply]
Oh, you mean texture encoding. I think when targeting ETC1S basisu encoder automatically determines if alpha is present based on image header (eg a PNG without transparency would be encoded without alpha slices); certainly let me know if that's not the case.

The encoding parameters can be selected via `-tq` command line argument in a simplified way; it maps each of 1-10 quality levels onto a preselected set of values that have been tuned to provide a slow increase in quality at the expense of bit rates for both ETC1S and UASTC, see this setup: https://github.com/zeux/meshoptimizer/blob/master/gltf/basisenc.cpp#L20-L65

**Reply by VanderSP** [reply]
1 - my question related to this workflow: packing gltf that contains all textures in png (even they only need output RGB) (also I never use etc1s)

2 - that script from basisu that make we encode via javascript have a quality level similar to yours (a simple setting), but toktx tool have extra fancy compression options that im not sure how your and basisu js cpp wasm implements

**Reply by VanderSP** [reply]
I got an interesting finding right now, yesterday i was using old tool (toktx), with ktx create it doesnt matter target rgb or rgba the output is same size
